grails-app/taglib/uk/org/nbn/biocache/hubs/OccurrenceTagLib.groovy

```diff
@@ -199,4 +199,99 @@ class OccurrenceTagLib extends au.org.ala.biocache.hubs.OccurrenceTagLib{
         }
     }
 
+    def wmsButton = { attrs ->
+        def targetSelector = attrs.targetSelector ?: 'body' // Default to body if no selector provided
+
+        // Output modal HTML directly to the page
+        out << g.render(template: '/occurrence/wmsModal')
```

Comment:
just wondering why the javascript can't be in the template too - instead of writing to a String and outputting it like below?

Reply:
moved JS to template - much better